src/stories/components/UserGreeting/UserGreeting.js

```diff
@@ -0,0 +1,54 @@
+import './styles.scss';
+
+import React, { useState, useEffect, useContext } from 'react';
+import UserServices from '../../../firebase/CRUDServices/userServices';
+
+export const UserContext = React.createContext();
+
+/**
+ * Component for user's greeting element.
+ *
+ * @component
+ * @return {object} (
+ *   <UserGreeting/>
+ * )
+ */
+export const UserGreeting = () => {
+  const service = UserServices.getInstance();
+  const user = useContext(UserContext);
+  const [logInState, setLogInState] = useState({ isLoggedIn: false });
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [error, setError] = useState('');
+
+  useEffect(async () => {
+    service.getCurrentUser(setLogInState).then((response) => {
+      if (response.errorCode) {
```

Comment:
I am not 100% sure, but I think we might have to be extra careful with expecting a response from the `getCurrentUser()` API. I looked at the code from that API and it looks like it doesn't return anything if the call to that API is successful (when the user is logged in and authenticated).

So maybe you could change line 26 to first check if there is a response before checking if there is an errorCode? Something like:

```
if (response && response.errorCode) {
  ...
}
```

I believe that statement will first check if the response object even exists, and only if it does exist then it will check for the errorCode.

By the way, this might be related to the Travis build fail that this PR is getting because I saw this message somewhere in there: `UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'errorCode' of undefined`.

Reply:
Thank you for checking on this. I didn't pay too much attention on what the API is returning when error is thrown.